lib/compress.js

```diff
@@ -3315,7 +3315,18 @@ merge(Compressor.prototype, {
     });
 
     AST_Definitions.DEFMETHOD("remove_initializers", function(){
-        this.definitions.forEach(function(def){ def.value = null });
+        var decls = [];
+        this.definitions.forEach(function(def) {
```

Comment:
This expensive walk can be avoided in most cases by wrapping it with:
```patch
+            def.value = null;
+            if (def.name instanceof AST_Destructuring) {
                   // walk...
```

Reply:
I have to clone the array out as total `length` may end up being different, so it's won't be as simple as `def.value = null` - but I see your point about performance.

Let me have a look and see what I can do.

Reply:
Done in https://github.com/mishoo/UglifyJS2/pull/2392/commits/1f823b52bc04e9cf343050d869231bfde97c990d